test/integration/integration_admin_test.cc

```diff
@@ -161,10 +161,37 @@ TEST_P(IntegrationAdminTest, Admin) {
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats?format=json",
                                                 "", downstreamProtocol(), version_);
   EXPECT_TRUE(response->complete());
-  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
   Json::ObjectSharedPtr statsjson = Json::Factory::loadFromString(response->body());
   EXPECT_TRUE(statsjson->hasObject("stats"));
   EXPECT_STREQ("application/json", ContentType(response));
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
+  uint64_t histogram_count = 0;
+  for (Json::ObjectSharedPtr obj_ptr : statsjson->getObjectArray("stats")) {
+    if (obj_ptr->hasObject("histograms")) {
+      histogram_count++;
+      const Json::ObjectSharedPtr& histograms_ptr = obj_ptr->getObject("histograms");
```

Comment:
I'm confused here: you are looking for 'histograms' plural so I was expecting  another loop over all the histograms.  Can you explain?

Reply:
There is a toplevel  "histograms" element that has supported_quantiles array and computed_quantiels, which is again an array of actual histogram computed values in the form of "name" and "values" (interval and cumulative) - the last array is repeated as many times as there are histograms"

Reply:
I'm still not entirely clear on the cardinality of variouis things from looking at the code.  If we actually add a second histogram and make the test expect 2?

Reply:
no. it is still 1. The computed_quantiles array size would be equivalent to the number of histograms.

Reply:
Can you please add a second histogram to this test, so computed_quantiles array has length 2?

Reply:
I am not adding any histograms in the test. I just grab whatever the output is from /stats?format=json and validate that it has histograms, it would definitely some (but non predictable). I am not setting up any thing for this test.

Reply:
Ah, very cool.  Let me rephrase my request.  Can you make a new test where you add some specific histogram data with more than one histogram, and check the JSON output?